core/src/main/scala/org/apache/spark/status/AppStatusStore.scala

```diff
@@ -94,6 +94,13 @@ private[spark] class AppStatusStore(
       }.toSeq
   }
 
+  def getJobIdsAssociatedWithStage(stageId: Int): Seq[Set[Int]] = {
```

Comment:
We don't need to fetch all the stage attempts, just the first of it is enough to get all the jobIds.

Reply:
Good point! Have taken care of it.